libopenage/gamestate/event/drag_select.h

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023-2023 the openage authors. See copying.md for legal info.
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+#include <string>
+
+#include "event/evententity.h"
+#include "event/eventhandler.h"
+
+
+namespace openage {
+
+namespace event {
+class EventLoop;
+class Event;
+class State;
+} // namespace event
+
+namespace gamestate::event {
+
+/**
+ * Drag select game entities.
+ */
+class DragSelectHandler : public openage::event::OnceEventHandler {
```

Comment:
i'm unsure if unit selections should be done with the event system - they could fit equally well just in the presenter layer? there things like the hud drawing pass take place anyway, and selecting units has nothing to do with the game simulation itself

Reply:
The problem is that selection depends a lot on the game simulation properties of entities, e.g.

* is the entity selectable?
* is the entity owned by the player using the drag select rectangle?
* can the entity be put in a selection queue? (buildings cannot for example)

Most of that is not available in the presenter layer, so making it an event makes more sense to me.

Reply:
and if we pass along a property like "selectable" to the presenter as a result of the simulation? selection with the mouse is a ui mechanic, which i think should be (apart from the flag) away from the simulation. i mean it doesn't hurt here but maybe later on we'll encounter a disadvantage or have trouble moving it then?

Reply:
I think this would be much harder now than leaving the current implementation intact. We would need to potentially mirror a lot of data and implement filters for deciding what game entity is selectable, since this has to be determined when the selection is made. And then, we might have to rewrite it a lot anyway to support index structures, selection areas, etc.

I would leave it as-is for now and revisit the selection mechanics when we do https://github.com/SFTtech/openage/issues/1542

Reply:
ok, then let's leave it and hope changing it for an improved method with index structures won't be too complicated :)